templates/pyomo/main.py

```diff
@@ -0,0 +1,158 @@
+"""
+Template for working with Pyomo.
+"""
+
+import argparse
+import json
+import sys
+from typing import Any
+
+import pyomo.environ as pyo
+
+# Duration parameter for the solver.
+SUPPORTED_PROVIDER_DURATIONS = {
+    "cbc": "sec",
+    "glpk": "tmlim",
+}
+
+
+# Status of the solver after optimizing.
+STATUS = {
+    pyo.TerminationCondition.feasible: "suboptimal",
+    pyo.TerminationCondition.infeasible: "infeasible",
+    pyo.TerminationCondition.optimal: "optimal",
+    pyo.TerminationCondition.unbounded: "unbounded",
+}
+
+
+def main() -> None:
+    """Entry point for the template."""
+
+    parser = argparse.ArgumentParser(description="Solve problems with Pyomo.")
+    parser.add_argument(
+        "-input",
+        default="",
+        help="Path to input file. Default is stdin.",
+    )
+    parser.add_argument(
+        "-output",
+        default="",
+        help="Path to output file. Default is stdout.",
+    )
+    parser.add_argument(
+        "-duration",
+        default=30,
+        help="Max runtime duration (in seconds). Default is 30.",
+        type=int,
+    )
+    args = parser.parse_args()
+
+    # Read input data, solve the problem and write the solution.
+    input_data = read_input(args.input)
+    log("Solving knapsack problem:")
+    log(f"  - items: {len(input_data.get('items', []))}")
+    log(f"  - capacity: {input_data.get('weight_capacity', 0)}")
+    log(f"  - max duration: {args.duration} seconds")
+    solution = solve(input_data, args.duration)
+    write_output(args.output, solution)
+
+
+def solve(input_data: dict[str, Any], duration: int) -> dict[str, Any]:
+    """Solves the given problem and returns the solution."""
+
+    # Creates the model.
+    model = pyo.ConcreteModel()
+    provider = "glpk"
```

Comment:
We might want to be more explicit about the fact we're defining a provider here, either in a comment in this file or in the template README. I could potentially see this being unintuitive to debug for CBC users depending on what error message they'd see when running this or if it simply would not behave as they'd expect.

Reply:
@cowanwalls Thanks. I changed the comment in the code, the error message and the README to make it more explicit that there are supported providers.